google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java

```diff
@@ -313,4 +324,26 @@ public void testGetServiceAccountProjectId_nonExistentFile() throws Exception {
 
     assertNull(ServiceOptions.getServiceAccountProjectId(credentialsFile.getPath()));
   }
+
+  @Test
```

Comment:
It's also worth testing the _negative_ case, e.g. when the response is invalid (like the Cox hijacked response). In particular this would reveal the NPE bug in the current code.